python-sdk/dev/scripts/pre_commit_readme_sync.py

```diff
@@ -0,0 +1,157 @@
+
+
+def get_table(records, fields, headings, alignment=None):
+    """
+    Generate a Doxygen-flavor Markdown table from records.
+
+    records -- Iterable.  Rows will be generated from this.
+    fields -- List of fields for each row.  Each entry may be an integer,
+        string or a function.  If the entry is an integer, it is assumed to be
+        an index of each record.  If the entry is a string, it is assumed to be
+        a field of each record.  If the entry is a function, it is called with
+        the record and its return value is taken as the value of the field.
+    headings -- List of column headings.
+    alignment - List of pairs alignment characters.  The first of the pair
+        specifies the alignment of the header, (Doxygen won't respect this, but
+        it might look good, the second specifies the alignment of the cells in
+        the column.
+
+        Possible alignment characters are:
+            '<' = Left align (default for cells)
+            '>' = Right align
+            '^' = Center (default for column headings)
+    """
+
+    num_columns = len(fields)
+    assert len(headings) == num_columns
+
+    # Compute the table cell data
+    columns = [[] for i in range(num_columns)]
+    for record in records:
+        for i, field in enumerate(fields):
+            columns[i].append(evalute_field(record, field))
+
+    # Fill out any missing alignment characters.
+    extended_align = alignment if alignment is not None else []
+    if len(extended_align) > num_columns:
+        extended_align = extended_align[0:num_columns]
+    elif len(extended_align) < num_columns:
+        extended_align += [("^", "<") for i in range(num_columns - len(extended_align))]
+
+    heading_align, cell_align = (x for x in zip(*extended_align))
+
+    field_widths = [len(max(column, key=len)) if len(column) > 0 else 0 for column in columns]
+    heading_widths = [max(len(head), 2) for head in headings]
+    column_widths = [max(x) for x in zip(field_widths, heading_widths)]
+
+    _ = " | ".join(["{:" + a + str(w) + "}" for a, w in zip(heading_align, column_widths)])
+    heading_template = "| " + _ + " |"
+    _ = " | ".join(["{:" + a + str(w) + "}" for a, w in zip(cell_align, column_widths)])
+    row_template = "| " + _ + " |"
+
+    _ = " | ".join([left_rule[a] + "-" * (w - 2) + right_rule[a] for a, w in zip(cell_align, column_widths)])
+    ruling = "| " + _ + " |"
+
+    result = []
+    result.append(heading_template.format(*headings).rstrip() + "\n")
+    result.append(ruling.rstrip() + "\n")
+
+    for row in zip(*columns):
+        result.append(row_template.format(*row).rstrip() + "\n")
+    return result
+
+
+def render_md_table(constants, constant_name) -> str:
+    """
+    create mark down table
+    """
+    return get_table(
+        headings=[constant_name], records=[[val] for val in getattr(constants, constant_name)], fields=[0]
+    )
+
+
+def get_line_numbers_of_section(file_path: str, heading: str):
```

Comment:
```suggestion
def get_line_numbers_of_section(file_path: Path, heading: str):
```
I think it would be great if we could have it as a Path instead of str

Reply:
Done.